### ISSUE TITLE
Component failing to parse

### ISSUE THREAD

**Issue by wswarr-spaulding** [Efferent-Health/HL7-dotnetcore#54]
The PV1 segment "PV1.7.1" for example doesn't parse out.  It behaves as if it thinks PV1.7 is not componentized.  

Here's a test that shows the issue (should be copiable to VS):

 [Test]
        public void ParsePV1()
        {
            string hl7File = 
                @"MSH|^~\&|EPIC||||20191107134803|ALEVIB01|ORM^O01|23|T|2.3|||||||||||
PID|1||10034926^^^NYU MRN^MRN||OSTRICH^DODUO||19820605|M||U|360 PARK AVE SOUTH^^NEW YORK^NY^10010^US^^^60|60|(615)613-2113^HOME^PH|||S|||999-99-9999|||U||N||||||||
PV1||O|NWSLED^^^NYULHLI^^^^^LI NW SLEEP DISORDER^^DEPID||||1447312459^WEINSTEIN^MICHAEL^^^^^^EPIC^^^^PNPI~WEINSM06^WEINSTEIN^MICHAEL^^^^^^KID^^^^KID|1447312459^WEINSTEIN^MICHAEL^^^^^^EPIC^^^^PNPI~WEINSM06^WEINSTEIN^MICHAEL^^^^^^KID^^^^KID|||||||||||496779945|||||||||||||||||||||||||20191107|||||||V
ORC|NW|162431500^EPC|30011922^EPC||Scheduled||^^^20191107134353^^R||20191107134806|ALEVIB01^ALEVIS^BRIAN^^||1447312459^WEINSTEIN^MICHAEL^^^^^^EPIC^^^^PNPI~WEINSM06^WEINSTEIN^MICHAEL^^^^^^KID^^^^KID|1084022001^^^1084022^^^^^LI NW SLEEP DISORDER&NYU WINTHROP SLEEP CENTER|(516)663-2834|||||||||||||||O|
OBR|1|162431500^EPC|30011922^EPC|IMG16525^SLEEP STUDY UNATTENDED^NAT^^SLEEP STUDY UNATTENDED|R|20191107134744|||||Ancillary Pe|||||1447312459^WEINSTEIN^MICHAEL^^^^^^EPIC^^^^PNPI~WEINSM06^WEINSTEIN^MICHAEL^^^^^^KID^^^^KID|(516)663-2834|NWSLED^^^^^^^^LI NW SLEEP DISORDER^^DEPID~LI NW SLEEP HOME STUDY||||||OT| Scheduled ||^^^20191107134355^^R|||||||||20191107134500||||||||IMG16525^SLEEP STUDY UNATTENDED^NAT^^SLEEP STUDY UNATTENDED|
NTE|1||Where will this exam be scheduled?->NYU Winthrop Sleep Disorders Center||
DG1|1|ICD-10-CM|W58.03XA^Crushed by alligator, initial encounter^ICD-10-CM|Crushed by alligator, initial encounter||^180;ORD
ZDS|1.2.840.114350.2.232.3.798268.2.162431500.1^EPIC^APPLICATION^DICOM";

            Message msg = new Message(hl7File);
            msg.ParseMessage();

            //This is OK
            string PID = msg.GetValue("PID.3.1");
            Assert.AreEqual("10034926", PID);

            //This fails, actually throws exception:
            try
            {
                string attendingDrId = msg.GetValue("PV1.7.1");
                Assert.AreEqual("1447312459", attendingDrId);
            }catch(Exception parseEx)
            {
                string err = parseEx.Message;
            }
        }

**Comment by jaime-olivares** [comment]
Hi win,
I cannot reproduce the issue, I have tried to test with the current source code and the current nuget package. The test always pass (test below). Is it possible that you have a strange character in the incoming message?

```
        [TestMethod]
        public void ReadComponentTest1()
        {
            var sampleMessage = @"MSH|^&|EPIC||||20191107134803|ALEVIB01|ORM^O01|23|T|2.3|||||||||||
PID|1||10034926^^^NYU MRN^MRN||OSTRICH^DODUO||19820605|M||U|360 PARK AVE SOUTH^^NEW YORK^NY^10010^US^^^60|60|(615)613-2113^HOME^PH|||S|||999-99-9999|||U||N||||||||
PV1||O|NWSLED^^^NYULHLI^^^^^LI NW SLEEP DISORDER^^DEPID||||1447312459^WEINSTEIN^MICHAEL^^^^^^EPIC^^^^PNPIWEINSM06^WEINSTEIN^MICHAEL^^^^^^KID^^^^KID|1447312459^WEINSTEIN^MICHAEL^^^^^^EPIC^^^^PNPIWEINSM06^WEINSTEIN^MICHAEL^^^^^^KID^^^^KID|||||||||||496779945|||||||||||||||||||||||||20191107|||||||V
ORC|NW|162431500^EPC|30011922^EPC||Scheduled||^^^20191107134353^^R||20191107134806|ALEVIB01^ALEVIS^BRIAN^^||1447312459^WEINSTEIN^MICHAEL^^^^^^EPIC^^^^PNPIWEINSM06^WEINSTEIN^MICHAEL^^^^^^KID^^^^KID|1084022001^^^1084022^^^^^LI NW SLEEP DISORDER&NYU WINTHROP SLEEP CENTER|(516)663-2834|||||||||||||||O|
OBR|1|162431500^EPC|30011922^EPC|IMG16525^SLEEP STUDY UNATTENDED^NAT^^SLEEP STUDY UNATTENDED|R|20191107134744|||||Ancillary Pe|||||1447312459^WEINSTEIN^MICHAEL^^^^^^EPIC^^^^PNPIWEINSM06^WEINSTEIN^MICHAEL^^^^^^KID^^^^KID|(516)663-2834|NWSLED^^^^^^^^LI NW SLEEP DISORDER^^DEPIDLI NW SLEEP HOME STUDY||||||OT| Scheduled ||^^^20191107134355^^R|||||||||20191107134500||||||||IMG16525^SLEEP STUDY UNATTENDED^NAT^^SLEEP STUDY UNATTENDED|
NTE|1||Where will this exam be scheduled?->NYU Winthrop Sleep Disorders Center||
DG1|1|ICD-10-CM|W58.03XA^Crushed by alligator, initial encounter^ICD-10-CM|Crushed by alligator, initial encounter||^180;ORD
ZDS|1.2.840.114350.2.232.3.798268.2.162431500.1^EPIC^APPLICATION^DICOM";

            var message = new Message(sampleMessage);
            message.ParseMessage();

            string PID = message.GetValue("PID.3.1");
            Assert.AreEqual("10034926", PID);

            string attendingDrId = message.GetValue("PV1.7.1");
            Assert.AreEqual("1447312459", attendingDrId);
        }        
```

Please elaborate what is the exception thrown.

**Comment by jaime-olivares** [comment]
Already diagnosed and fully-solved the issue:
The PV1 segment has a repeated field and the default indexing form `message.GetValue("PV1.7.1")` was failing. When a field is repeated, you had to use the form recommended in the README: 

`message.Segments("PV1")[0].Fields(7).Repetitions(1).Components(1)`;

Now there are a couple of new indexing alternatives producing the same result:
```
message.GetValue("PV1.7.1");    // Will take the first repetition by default
message.GetValue("PV1.7(1).1");
message.GetValue("PV1.7[1].1");
```

The change is already published in nuget as version 2.17.0: 
https://www.nuget.org/packages/HL7-dotnetcore/2.17.0
